File: src/Cxx/Utilities/ReportRenderWindowCapabilities.cxx
#include <vtkSmartPointer.h>

#include <vtkConeSource.h>
#include <vtkPolyData.h>
#include <vtkPolyDataMapper.h>
#include <vtkActor.h>
#include <vtkRenderWindow.h>
#include <vtkRenderer.h>

int main(int, char *[])
{
  //Create a cone
  vtkSmartPointer<vtkConeSource> coneSource =
    vtkSmartPointer<vtkConeSource>::New();
  coneSource->Update();

  //Create a mapper and actor
  vtkSmartPointer<vtkPolyDataMapper> mapper =
    vtkSmartPointer<vtkPolyDataMapper>::New();
  mapper->SetInputConnection(coneSource->GetOutputPort());

  vtkSmartPointer<vtkActor> actor =
    vtkSmartPointer<vtkActor>::New();
  actor->SetMapper(mapper);

  //Create a renderer and render window<
  vtkSmartPointer<vtkRenderer> renderer =
    vtkSmartPointer<vtkRenderer>::New();
  vtkSmartPointer<vtkRenderWindow> renderWindow =
    vtkSmartPointer<vtkRenderWindow>::New();
  renderWindow->AddRenderer(renderer);

  //Add the actors to the scene
  renderer->AddActor(actor);

  //Render
  renderWindow->Render();

  std::cout << renderWindow->ReportCapabilities();

  return EXIT_SUCCESS;
}

File: src/Java/Utilities/ReportRenderWindowCapabilities.java
import vtk.vtkActor;
import vtk.vtkNativeLibrary;
import vtk.vtkPolyDataMapper;
import vtk.vtkRenderWindow;
import vtk.vtkRenderer;
import vtk.vtkConeSource;

public class ReportRenderWindowCapabilities 
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------

  public static void main(String args[]) 
  {
   //Create a cone
   vtkConeSource coneSource = new vtkConeSource();
   coneSource.Update();
   
   //Create a mapper and actor
   vtkPolyDataMapper mapper = new vtkPolyDataMapper();
   mapper.SetInputConnection(coneSource.GetOutputPort());
   
   vtkActor actor = new vtkActor();
   actor.SetMapper(mapper);
   
   // Create the renderer, render window
   vtkRenderer ren = new vtkRenderer();
   vtkRenderWindow renWin = new vtkRenderWindow();
   renWin.AddRenderer(ren);
   
   ren.AddActor(actor);
   
   renWin.Render();
   
   System.out.println(renWin.ReportCapabilities());
  }
}
